__tests__/app.test.js

```diff
@@ -36,7 +36,214 @@ describe("GET/api", () => {
+        const { text } = error;
+        expect(text).toBe("Not found");
+      });
+  });
+  test("404: When the request is a number out of the range it replies with Not found", () => {
+    return request(app)
+      .get("/api/articles/11111111111111111111/comments")
+      .expect(404)
+      .then(({ body }) => {
+        expect(body.msg).toBe("Not found");
+      });
+  });
+});
+
+describe("/api/articles/:article_id/comments", () => {
+  test("201: posts a comment to the article on the correct id", () => {
+    const testComment = {
+      username: "butter_bridge",
+      body: "testing",
+    };
+
+    return request(app)
+      .post("/api/articles/1/comments")
+      .send(testComment)
+      .expect(201)
+      .then((res) => {
+        const newComment = res.body;
+        expect(newComment).toHaveProperty("comment_id", 19);
+        expect(newComment).toHaveProperty("body", "testing");
+        expect(newComment).toHaveProperty("article_id", 1);
+        expect(newComment).toHaveProperty("author", "butter_bridge");
+        expect(newComment).toHaveProperty("votes", 0);
+        expect(newComment).toHaveProperty("created_at", expect.any(String));
+      });
+  });
+  test("400: Responds with a bad request when trying to post a comment with nothing in the column when it is marked as NOT NULL", () => {
+    const testComment = {
+      username: null,
+      body: "testing",
+    };
+    return request(app)
+      .post("/api/articles/1/comments")
+      .send(testComment)
+      .expect(400)
+      .then(({ body }) => {
+        const { msg } = body;
+        expect(msg).toBe("Bad Request");
+      });
+  });
+  test("400: Responds with a bad request when trying to post a comment from a user that does not exist", () => {
+    const testComment = {
+      username: "Test User",
+      body: "testing",
+    };
+    return request(app)
+      .post("/api/articles/1/comments")
+      .send(testComment)
+      .expect(400)
+      .then(({ body }) => {
+        const { msg } = body;
+        expect(msg).toBe("Bad Request");
+      });
+  });
+  test("400: Responds with a bad request when trying to post a comment where the user name is not a string", () => {
+    const testComment = {
+      username: 25,
+      body: 25,
+    };
+    return request(app)
+      .post("/api/articles/1/comments")
+      .send(testComment)
+      .expect(400)
+      .then(({ body }) => {
+        const { msg } = body;
+        expect(msg).toBe("Bad Request");
+      });
+  });
+  test("404: responds with Not found if trying to post to an article ID that doesn't exist", () => {
+      const testComment = {
+          username: "butter_bridge",
+          body: "testing",
+        };
+        
+        return request(app)
+        .post("/api/articles/999/comments")
+        .send(testComment)
+        .expect(404)
+        .then(({ error }) => {
+            const { text } = error;
+            expect(text).toBe("Not found");
+          });
+    });
+})
+
```

Comment:
Nice testing so far, but could you just add the following two please?
- A status 201 that ignores any unnecessary properties (i,e. if we were to send an extra key as well as username and body)
- Status 400 for invalid ID (ie. /api/articles/banana/comments)